### PR TITLE
Implementing Matte Anything

### PR DESCRIPTION
**Pull request by chloedia**
Hi, here is a PR to indicate I am working on implementing Matte Anything on refiners.
Here is the steps I identified : 
Step 1 : Implement the refiner version of ViT Matte
- [X] Implement the ViT Backbone and validate correct state_dict conversion;
- [X] Implement the Decoder and validate correct state_dict conversion;
- [X] Implement the VitMatte and validate correct state_dict conversion;
- [X] Write a conversion script for VitMatte (convert_ViTMatte.py)

Step 2: Implement Grounding Dino

- [ ] Implement the Text Backbone
- [ ] Implement the Image Backbone
- [ ] Implement the Feature Enhancer
- [ ] Implement the Decoder Layer

Step 3: Implement the Matte Anything

- [ ] Set foreground and background points (?);*
- [ ] Give foreground caption and use **Grounding Dino** to detect the objects;**
- [ ] Give the fg and bg points + the bounding boxes as input to **SAM** and get the segmentation;
- [ ] Combine all the masks;
- [ ]  Generate alpha matte : 
    - [ ] Generate the trimap;
    - [ ] Using grounding Dino, detect transparent objects (given specific set of words) and annotate on the trimap (give a specific value 0.5 - transparency);
- [ ] Use ViT Matte to extract a finer segmentation of the foreground;

Here are my first questions/ assumptions : 
1. Do we assume fg and bg points are given in entry ? (no UI)
2. Do we assume the fg caption are given in entry ? (no UI)

### CODE REVIEW

**Comment by brycedrennan** [comment]
Looks like the commits somehow got tangled. Let me know if you want help resolving that. I'm on the discord right now `imaginairybryce`
